components/match2/wikis/sideswipe/match_group_input_custom.lua

```diff
@@ -0,0 +1,532 @@
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+function mapFunctions.getExtraData(map)
+	map.extradata = {
+		ot = map.ot,
+		otlength = map.otlength,
+		comment = map.comment,
+		header = map.header,
+		--the following is used to store 'mapXtYgoals' from LegacyMatchLists
+		t1goals = map.t1goals,
+		t2goals = map.t2goals,
+	}
+	return map
+end
+
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+	for scoreIndex = 1, _MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				score = tonumber(score)
+				obj.status = _STATUS_HAS_SCORE
+				obj.score = score
+				obj.index = scoreIndex
+			elseif Table.includes(_ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+				obj.index = scoreIndex
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+	local isFinished = map.finished
+	if not Logic.isEmpty(isFinished) then
+		isFinished = Logic.readBool(isFinished)
+	else
+		isFinished = not Logic.readBool(map.unfinished)
+	end
+	if isFinished and not Logic.isEmpty(indexedScores) then
+		map.winner = mapFunctions.getWinner(indexedScores)
+	end
+
+	return map
+end
+
+function mapFunctions.getWinner(indexedScores)
+	table.sort(indexedScores, mapFunctions.mapWinnerSortFunction)
+	return indexedScores[1].index
+end
+
+function mapFunctions.mapWinnerSortFunction(op1, op2)
+	local op1norm = op1.status == _STATUS_HAS_SCORE
+	local op2norm = op2.status == _STATUS_HAS_SCORE
+	if op1norm then
+		if op2norm then
+			return tonumber(op1.score) > tonumber(op2.score)
+		else return true end
+	else
+		if op2norm then return false
+		elseif op1.status == _STATUS_DEFAULT_WIN then return true
+		elseif Table.includes(_ALLOWED_STATUSES, op1.status) then return false
+		elseif op2.status == _STATUS_DEFAULT_WIN then return false
+		elseif Table.includes(_ALLOWED_STATUSES, op2.status) then return true
+		else return true end
+	end
+end
+
+function mapFunctions.getTournamentVars(map)
+	map.mode = Logic.emptyOr(map.mode, Variables.varDefault('tournament_mode', '3v3'))
+	map.type = Logic.emptyOr(map.type, Variables.varDefault('tournament_type'))
+	map.tournament = Logic.emptyOr(map.tournament, Variables.varDefault('tournament_name'))
+	map.tickername = Logic.emptyOr(map.tickername, Variables.varDefault('tournament_ticker_name'))
+	map.shortname = Logic.emptyOr(map.shortname, Variables.varDefault('tournament_shortname'))
+	map.series = Logic.emptyOr(map.series, Variables.varDefault('tournament_series'))
+	map.icon = Logic.emptyOr(map.icon, Variables.varDefault('tournament_icon'))
+	map.icondark = Logic.emptyOr(map.iconDark, Variables.varDefault('tournament_icon_dark'))
+	map.liquipediatier = Logic.emptyOr(map.liquipediatier, Variables.varDefault('tournament_tier'))
```

Comment:
Same as above

Reply:
Isn't this a duplicate method? If so wouldn't it be better to remove it since everything is already set via the above method (lines 120-132)

Reply:
One is used for maps, one for matches. Unifying is indeed the correct option.